src/rgw/rgw_file.h

```diff
@@ -1026,7 +1041,7 @@ namespace rgw {
 	  fh_cache.insert_latched(fh, lat, RGWFileHandle::FHCache::FLAG_UNLOCK);
 	  get<1>(fhr) |= RGWFileHandle::FLAG_CREATE;
 	  /* ref parent (non-initial ref cannot fail on valid object) */
-	  if (! parent->is_root()) {
+	  if (! parent->is_mount()) {
```

Comment:
@mattbenjamin iiuc, is_root() aims to check for the real "/", and is_mount() check for a root_fh which may points to the "/" or a bucket, and root_fh is not supposed to be in LRU, so ref() & unref() should adopt is_mount().